test/spmd/tensor/test_megatron_example.py

```diff
@@ -129,6 +187,44 @@ def test_simple_megatron_e2e(self):
         output_tp = model_tp(inp)
```

Comment:
I would just do a 2 step loop to check that we can do a FW + BW post optimizer.step(). That way we can check if the grads match as well as the params for both TP and non-TP

Reply:
I think we are already doing that?

Reply:
I am suggesting the following for the sake of readability:
1. Initialize model
2. Initialize optimizer
3. for each step in the loop
-> check initial params match
-> check grads match
-> check params match post step()

Right now by running a FW post step() call, you are trying to verify the same criteria as the above FW pass.

Reply:
I see. Got it.